swagger.yaml

```diff
@@ -1,42 +1,292 @@
     url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+
 basePath: "/v1"
+
 schemes:
   - http
+
 tags:
-  - name: "hello"
+  - name: "image"
+
 paths:
-  /hello:
+
+  /images:
     get:
       tags:
-        - hello
-      summary: Example hello world endpoint
-      description: Returns an example payload.
+      - "image"
+      summary: "Get images filtered by collection id"
+      description: "Returns a list of images metadata filtered by a compulsory query parameter defining the collection ID"
+      parameters:
+      - $ref: '#/parameters/collection_id'
       produces:
-        - application/json
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
       responses:
         200:
-          description: OK
+          description: "A json object containing images which correspond to the provided collection ID"
+          schema:
+            $ref: '#/definitions/Images'
+        400:
+          description: "Invalid request, collection id was incorrect"
+        401:
+          description: "Unauthorised to view images metadata"
+        404:
+          $ref: '#/responses/NotFound'
+        500:
+          $ref: '#/responses/InternalError'
+    post:
+      tags:
+      - "image"
+      summary: "Create a new image metadata entry"
+      description: "Creates a new image metadata entry corresponding to the provided body in this request."
+      parameters:
+      - $ref: '#/parameters/image'
+      produces:
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
+      responses:
+        201:
+          description: "The image metadata was correctly created and a json object containing the new image information is returned, including a newly generated id"
           schema:
-            $ref: "#/definitions/HelloResponse"
+            $ref: '#/definitions/Images'
+        400:
+          description: |
+            Invalid request, reasons can be one of the following:
+              * collection id was incorrect
+              * malformed body
+              * provided image had an invalid parameter
+        401:
+          description: "Unauthorised to create images metadata"
+        409:
+          description: "Cannot create an image for the provided collection ID because it is already published"
+        404:
+          $ref: '#/responses/NotFound'
+        500:
+          $ref: '#/responses/InternalError'
+
+  /images/{image_id}:
+    get:
+      tags:
+      - "image"
+      summary: "Get an image metadata by its id"
+      description: "Returns an image metadata whose id matches the id provided as the last element of the path"
+      parameters:
+      - $ref: '#/parameters/image_id'
+      produces:
+      - "application/json"
+      security:
+      - FlorenceAPIKey: []
+      responses:
+        200:
+          description: "A json with the requested image metadata"
+          schema:
+            $ref: '#/definitions/Image'
+        400:
```

Comment:
We probably won't need a `400` here as if the image id does not exist/invalid/etc. we will be returning a `404`.